scripts/prepare-subcharts

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+set -e
+
+# Pull in the subcharts required by a chart and apply any patches on subcharts with fixed version numbers
+
+if [[ -z $1 ]]; then
+	echo "No directory provided to pull in subcharts"
+	exit 1
+fi
+
+f=$1
+
+if ! [[ -f ${f}/charts/requirements.yaml ]]; then
+	echo "Chart ${CHART} has no dependencies; skipping step to pull in dependencies"
+	exit 0
+fi
+
+# Apply the patch on the requirements.yaml
+for file in $(find ./${f} -type f -name "*.patch"); do
+	awk '/diff.*requirements.yaml/{flag=1;next}/diff/{flag=0}flag' $file >> ${f}/requirements.patch
+done
+if [[ -f ${f}/requirements.patch ]] && [[ -s ${f}/requirements.patch ]]; then
+	patch -E -p3 -d ${f}/charts < ${f}/requirements.patch
+fi
+
+# Pull in any dependencies that are stored as package.yamls or in the overlay/ folder
+overlaid_dependencies=()
+mirrored_dependencies=()
+i=0
+while ! [[ -z $(yq r ${f}/charts/requirements.yaml "dependencies[${i}]") ]]; do
+	repo=$(yq r ${f}/charts/requirements.yaml "dependencies[${i}].repository")
+	if [[ ${repo} == file://* ]]; then
+		charts_path=${repo/file:\/\//${f}/charts/}
+		if [[ -d ${charts_path} ]]; then
+			# Chart already exists
+			((i=i+1))
+			continue
+		fi
+		if [[ -f ${charts_path%/*}/package.yaml ]]; then
+			# Prepare the subchart and pull it in
+			subchart=$(basename -- ${charts_path%/*})
+			mirrored_dependencies+=("${subchart}")
+			prepare_subchart () {
+				local CURRENT_CHART=${CHART}
+				local CURRENT_INDEX=${i}
+				CHART=${subchart} ./scripts/prepare
```

Comment:
Yes; if we are running with `CHART=` or when `CHART` is unset, this would double call the scripts / prepare  if it's already another chart in the repo  (i.e. for our CI scripts). For example, this would prepare something like `rancher-kiali-server` twice.

However, for the average dev that uses `CHART=<chart-name>` when running the scripts, this recursive prepare is required in order to ensure any subcharts are also prepared / packaged (i.e. running `make CHART=rancher-istio prepare` needs to prepare `rancher-kiali-server` first).

One thing to note: there's no explicit check here about a circular dependency on prepares (i.e. `rancher-istio` depends on `rancher-kiali-server` depends on `rancher-istio` resulting in an eternal loop of prepares), but the assumption is that the developer won't run into this issue when designing charts.